hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/crypto/airdrops/TokenAirdropSuite.java

```diff
@@ -0,0 +1,821 @@
+                        getAccountBalance(RECEIVER_WITH_FREE_AUTO_ASSOCIATIONS).hasTokenBalance(NON_FUNGIBLE_TOKEN, 1),
+                        getAccountBalance(RECEIVER_WITH_0_AUTO_ASSOCIATIONS).hasTokenBalance(NON_FUNGIBLE_TOKEN, 0),
+                        getAccountBalance(RECEIVER_WITHOUT_FREE_AUTO_ASSOCIATIONS)
+                                .hasTokenBalance(NON_FUNGIBLE_TOKEN, 0));
+    }
+
+    @HapiTest
+    @DisplayName("in pending state, should prepay hbar custom fee")
+    final Stream<DynamicTest> airdropFungibleWithFixedHbarCustomFee() {
+        final long hbarFee = 1_000L;
+        final long tokenTotal = 1_000L;
+        final String hbarCollector = "hbarCollector";
+        return defaultHapiSpec("with fixed Hbar CustomFee")
+                .given(
+                        cryptoCreate(hbarCollector).balance(0L),
+                        cryptoCreate(OWNER).balance(ONE_MILLION_HBARS),
+                        cryptoCreate(RECEIVER_WITH_0_AUTO_ASSOCIATIONS),
+                        cryptoCreate(TOKEN_TREASURY),
+                        tokenCreate(FUNGIBLE_TOKEN)
+                                .treasury(TOKEN_TREASURY)
+                                .tokenType(TokenType.FUNGIBLE_COMMON)
+                                .initialSupply(tokenTotal)
+                                .withCustom(fixedHbarFee(hbarFee, hbarCollector)),
+                        tokenAssociate(OWNER, FUNGIBLE_TOKEN),
+                        cryptoTransfer(moving(1000, FUNGIBLE_TOKEN).between(TOKEN_TREASURY, OWNER)))
+                .when(tokenAirdrop(moving(1, FUNGIBLE_TOKEN).between(OWNER, RECEIVER_WITH_0_AUTO_ASSOCIATIONS))
+                        .fee(ONE_HUNDRED_HBARS)
+                        .via("transferTx")
+                        .payingWith(OWNER))
+                .then(
+                        // assert balances
+                        getAccountBalance(RECEIVER_WITH_0_AUTO_ASSOCIATIONS).hasTokenBalance(FUNGIBLE_TOKEN, 0),
+                        getAccountBalance(hbarCollector).hasTinyBars(hbarFee),
+                        withOpContext((spec, log) -> {
+                            final var record = getTxnRecord("transferTx");
+                            allRunFor(spec, record);
+                            final var txFee = record.getResponseRecord().getTransactionFee();
+                            // the token should not be transferred but the custom fee should be charged
+                            final var ownerBalance = getAccountBalance(OWNER)
+                                    .hasTinyBars(ONE_MILLION_HBARS - (txFee + hbarFee))
+                                    .hasTokenBalance(FUNGIBLE_TOKEN, 1000);
+                            allRunFor(spec, ownerBalance);
+                        }));
+    }
+
+    @HapiTest
+    @DisplayName("in pending state, should prepay hts custom fee")
+    final Stream<DynamicTest> transferNonFungibleWithFixedHtsCustomFees2Layers() {
```

Comment:
We should follow the checklist Michael [added](https://github.com/hashgraph/hedera-services/blob/develop/hedera-node/test-clients/README.md#the-hapitest-checklist) .